docs/concepts/composition.md

```diff
@@ -130,6 +130,9 @@ This can be surprising, so check the manifests of the included environments and
 It's reasonable to assume that the manifests of the included environments will change over time and at some point you will want to bring in the latest versions of those manifests.
 This is accomplished with the `flox include upgrade` command.
 
-Running this command will check each of the included environments and determine if there were changes to their manifests.
-If the changes to the manifest have been built by the included environment, then the new manifest will be included.
+Running this command will check each of the included environments and check if the environments have changed.
+In order for changes to be pulled in, the included environment must have a lockfile corresponding to the latest changes to the manifest.
```

Comment:
```suggestion
In order for changes to be pulled in, the manifest and lockfile of the the included environment must be in sync with one another (it's possible for them to get out of sync if you edit `manifest.toml` with your editor rather than `flox edit`).
```

Suggestion: this reads a little smoother to me